packages/plugin-core/src/test/suite-integ/workspaceActivator.test.ts

```diff
@@ -0,0 +1,153 @@
+import { WorkspaceService } from "@dendronhq/engine-server";
+import { ENGINE_HOOKS } from "@dendronhq/engine-test-utils";
+import { ExtensionProvider } from "../../ExtensionProvider";
+import { describeMultiWS } from "../testUtilsV3";
+import { describe } from "mocha";
+import { expect } from "../testUtilsv2";
+import { trackTopLevelRepoFound } from "../../workspace/workspaceActivator";
+import sinon from "sinon";
+import _ from "lodash";
+
+suite("workspaceActivator", function () {
+  describe("trackTopLevelRepoFound", () => {
```

Comment:
can we add a test case when there is no repo found?

Reply:
I initially omitted the no repo case because I assumed this would all be covered by unit tests in `git.spec.ts`, and because this integ test was already feeling a bit redundant.

> covered by unit tests in `git.spec.ts`

This was not the case, so I added a few unit tests that needs to be covered in order to verify `trackTopLevelRepoFound` works.

> integ test was already feeling a bit redundant

and also changed my mind on this after realizing we don't have good coverage in unit tests (at least the ones related to the `Git` class). So I also added the `no repo` case in the integ tests.